### PR TITLE
Project status dialog improvements

### LINKED ISSUE #172: Better project status report
Right now the status is only reported in a single message box as text, which is not always easy to read, especially if there are many files modified. It would be good to create a dialog that would incllude a tree view with the files and there would be icons next to file names telling whether the change was addition/removal/modification. For geopackages, we should also show detailed information from geodiff - which tables are changed (and how many inserted/deletes/updates) - this could be done with child items of the geopackage file item.

### PR DESCRIPTION
**Pull request by sklencar**
Remarks: 
- consider to use background colors according category (removed, added, modified)
- icon for rename and edit is currently the same

Preview of the dialog with warning/info text (text forced to show therefore the end of text has missing list of files
Screenshot 2020-11-10 at 14 23 26
)

closes #172

### CODE REVIEW

**Comment by saberraz** [comment]
Resolves: https://github.com/lutraconsulting/qgis-mergin-plugin/issues/172

File: Mergin/plugin.py [hunk 1]
```diff
@@ -294,44 +295,12 @@ def project_status(self):
...
-
-            msg = ''
-            if sum(len(v) for v in pull_changes.values()):
-                msg += pull_msg + "\n"
-            if sum(len(v) for v in push_changes.values()):
-                msg += push_msg
-                msg += pretty_summary(push_changes_summary)
-                msg += f"\nUnable to compare some of the modified local files with their server version - we will have to upload the whole file (history of the files will be lost): {files_to_replace}" if files_to_replace else ""
-                if not self._have_writing_permissions():
-                    msg += f"\n\nWARNING: You don't have writing permissions to this project. Changes won't be synced!"
```
Comment (wonder-sk) [inline Mergin/plugin.py:330]: This check for writing permissions somehow got lost

File: Mergin/plugin.py [hunk 2]
```diff
@@ -294,44 +295,12 @@ def project_status(self):
...
-
-            files_to_replace = ", ".join([file["path"] for file in push_changes["updated"] if "diff" not in file and ".gpkg" in file['path']])
-
-            msg = ''
-            if sum(len(v) for v in pull_changes.values()):
-                msg += pull_msg + "\n"
-            if sum(len(v) for v in push_changes.values()):
-                msg += push_msg
-                msg += pretty_summary(push_changes_summary)
-                msg += f"\nUnable to compare some of the modified local files with their server version - we will have to upload the whole file (history of the files will be lost): {files_to_replace}" if files_to_replace else ""
```
Comment (wonder-sk) [inline Mergin/plugin.py:328]: This warning got lost - if a diffable file does not have diff info from geodiff, we should show a warning as a child item of such file

File: Mergin/project_status_dialog.py [hunk 3]
```diff
@@ -37,12 +38,31 @@ def __init__(self, pull_changes, push_changes, push_changes_summary, parent=None
...
+    def _get_info_text(self, push_changes, has_write_permissions):
+        msg = ""
+        if not has_write_permissions:
+            msg += f"WARNING: You don't have writing permissions to this project. Changes won't be synced!\n"
+
+        files_to_replace = ", ".join(
+            [file["path"] for file in push_changes["updated"] if "diff" not in file and ".gpkg" in file['path']])
+        if files_to_replace:
+            msg += f"\nUnable to compare some of the modified local files with their server version - we will have to " \
+               f"upload the whole file (history of the files will be lost): {files_to_replace}"
```
Comment (wonder-sk) [inline Mergin/project_status_dialog.py:62]: Thanks for the update... Can we improve this a little bit:

- the info text would say `WARNING: Unable to compare some of the modified files with their server version - their history will be lost if uploaded.`  - without the list of files in the label text
- if a particular gpkg/sqlite file has geodiff failed, a child item for that file would say e.g. `Unable to detect changes`

File: Mergin/project_status_dialog.py [hunk 4]
```diff
@@ -37,12 +38,31 @@ def __init__(self, pull_changes, push_changes, push_changes_summary, parent=None
...
 
         self.resize(640, 640)
 
+    def _get_info_text(self, push_changes, has_write_permissions):
+        msg = ""
+        if not has_write_permissions:
+            msg += f"WARNING: You don't have writing permissions to this project. Changes won't be synced!\n"
+
+        files_to_replace = ", ".join(
+            [file["path"] for file in push_changes["updated"] if "diff" not in file and ".gpkg" in file['path']])
```
Comment (wonder-sk) [inline Mergin/project_status_dialog.py:59]: `".gpkg" in file['path']`  -- don't we have some function for that? (is_versioned_file() or something similar?)

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**